### PR TITLE
Relax volume validation requirements

### PR DESCRIPTION
**Pull request by caipre**
This commit adds support for mounting named volumes by relaxing the
validation performed by the `JobValidator`. This allows jobs to use
options for volumes managed by a particular driver, such as:

```
docker volume create --driver local \
   --opt type=btrfs --opt device=/dev/sda2 \
   myvolume
```

The above volume can then be mounted into a container using the
following helios job configuration (full configuration omitted):

```
{
  ...
  "volumes": {
    "/mnt/myvol": "myvolume"
  }
}
```

@mattnworb @davidxia

### CODE REVIEW

File: docs/user_manual.md [hunk 1]
```diff
@@ -421,13 +421,22 @@ container.
...
-The container path can be suffixed with "rw" or "ro" to create a read-write or
-read-only volume, respectively.
+Specify a single path to create a data volume, a container path and a source
+path to mount a directory from the host, or a container path and a name to mount
+a named volume. The named volume will be created by docker if it does not exist;
+see [volumes][] in the Docker reference for full details. The container path can
+be suffixed with "rw" or "ro" to create a read-write or read-only volume mount,
+respectively.
+
+Format: `[container-path]:[rw|ro]:[host-path|volume-name]`.
```
Comment (mattnworb) [inline docs/user_manual.md]: The format here doesn't match what is mentioned in the bullet points below, since the latter is representing it as a JSON object. Could you make this line match the below (i.e. with double quotes)?

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java [hunk 2]
```diff
@@ -161,13 +161,10 @@ public JobValidator(final boolean shouldValidateJobHash,
         errors.add("Volume path is not absolute: " + path);
         continue;
       }
-      if (!isNullOrEmpty(source) && !source.startsWith("/")) {
-        errors.add("Volume source is not absolute: " + source);
-        continue;
-      }
```
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: Should we still do some validation of `String source = entry.getValue()` ? For instance I'd expect that if it contained a `/`, it would need to be an absolute path. I'd imagine that for the other types of volumes a `/` would never be allowed (although trying to reimplement docker's validation logic would be overkill)

File: helios-client/src/test/java/com/spotify/helios/common/JobValidatorTest.java [hunk 3]
```diff
@@ -169,6 +169,7 @@ public void testValidHostnamesPass() {
   @Test
   public void testValidVolumesPass() {
     final Job j = Job.newBuilder().setName("foo").setVersion("1").setImage("foobar").build();
+    assertThat(validator.validate(j.toBuilder().addVolume("/foo", "bar").build()), is(empty()));
```
Comment (mattnworb) [inline helios-client/src/test/java/com/spotify/helios/common/JobValidatorTest.java]: could you add a test of `.addVolume("/foo", "")` too? (if it is not covered elsewhere)

File: helios-services/src/main/java/com/spotify/helios/agent/TaskConfig.java [hunk 4]
```diff
@@ -112,7 +112,7 @@ public ContainerConfig containerConfig(final ImageInfo imageInfo,
     builder.hostname(job.getHostname());
     builder.env(containerEnvStrings());
     builder.exposedPorts(containerExposedPorts());
-    builder.volumes(volumes());
+    builder.volumes(volumes().keySet());
```
Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/agent/TaskConfig.java]: why is this changed, wouldn't it lose the source part of the volume specification?

**Comment by caipre** [comment]
> Do you know what would happen if a Helios job had a volume spec like "/foo/bar":"named-volume" and that volume did not exist?

When I tried this locally Docker created the volume and mounted it as expected.

File: helios-services/src/main/java/com/spotify/helios/agent/TaskConfig.java (hunk 4 above)
Comment (caipre) [reply helios-services/src/main/java/com/spotify/helios/agent/TaskConfig.java]: The method accepting a `Map<String, Map>` [was deprecated](https://github.com/spotify/docker-client/blob/353417a693174296b4cba538e8c7ef5ae177b3e9/src/main/java/com/spotify/docker/client/messages/ContainerConfig.java#L278-L286).
